src/parse.rs

```diff
@@ -305,7 +305,13 @@ fn parse_fn(item: &ast::Item) -> Result<Option<String>, Error> {
         let has_args = !fn_decl.inputs.is_empty();
 
         for arg in &fn_decl.inputs {
-            let arg_name = print::pprust::pat_to_string(&*arg.pat);
+            use syntax::ast::{PatIdent, BindingMode};
+            let arg_name = match arg.pat.node {
+                PatIdent(BindingMode::ByValue(_), ref ident, None) => {
+                    ident.node.name.to_string()
+                }
+                _ => panic!("hi")
```

Comment:
This is an unreachable arm isn't it? If so could you change that to something along the lines of

```rust
_ => unreachable!("cheddar should have already check that the arguments are  by-value patterns"),
```

please.